nncf/torch/compression_method_api.py

```diff
@@ -80,11 +82,50 @@ def get_state(self) -> None:
         """
 
 
+class PTCompressionState(CompressionState):
+    """
+    Entire compression state, containing state of `NNCFNetwork`, and builder and controller states.
+    This state can be used to resume compression via `compression_state` parameter of `create_compressed_model`
+    """
+    MODEL_STATE_ATTR = 'nncf_model_state'
+
+    def __init__(self, compression_setups: List[CompressionSetup] = None, model_state: Dict[str, torch.Tensor] = None):
+        super().__init__(compression_setups)
+        self._model_state = model_state
+
+    @property
+    def model_state(self) -> Dict[str, torch.Tensor]:
+        return self._model_state
+
+    def get_state(self) -> Dict:
+        """
+        :return: PT-specific representation of the object for saving to the checkpoint
+        """
+        state = super().get_state()
+        state[self.MODEL_STATE_ATTR] = self._model_state
+        return state
+
+    def load_state(self, state: Dict):
+        """
+        Loads state to the object
+        :param state: Output of `get_state()` method.
+        """
+        super().load_state(state)
+        self._model_state = state[self.MODEL_STATE_ATTR]
+
+
 class PTCompressionAlgorithmController(BaseCompressionAlgorithmController):
     """Serves as a handle to the additional modules, parameters and hooks inserted
     into the original uncompressed model in order to enable algorithm-specific compression.
     Hosts entities that are to be used during the training process, such as compression scheduler and
     compression loss."""
+    COMPRESSION_STAGE_ATTR = 'compression_stage'
+    COMPRESSION_LEVEL_ATTR = 'compression_level'
+    SCHEDULER_ATTR = 'scheduler'
```

Comment:
The thing is that, eventually for TF and PT, we need a dictionary with Python objects, not a JSON string for each state.
Otherwise, serialization wouldn't be generalizable - having python object we have a choice to which format to serialize: json (TF), yaml, pickle by torch.save (current approach for PT).

These state objects would only complicate things. It would be a wrapper on top of existing dict, requiring to create a new class, pass arguments for init, call special method to get a wrapped dict with Python objects.